app/src/main/java/org/ergoplatform/android/wallet/CreateWalletFragmentDialog.kt

```diff
@@ -42,7 +50,17 @@ class CreateWalletFragmentDialog : FullScreenFragmentDialog() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        binding.tvMnemonic.text = Mnemonic.generateEnglishMnemonic()
+        binding.tvMnemonic.text =
+            if (savedInstanceState?.containsKey(BUNDLE_ID_MNEMONIC) == true)
+                savedInstanceState.getString(BUNDLE_ID_MNEMONIC)
+            else
+                Mnemonic.generateEnglishMnemonic()
+
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString(BUNDLE_ID_MNEMONIC, binding.tvMnemonic.text.toString())
```

Comment:
So, this means the mnemonic is leaking to the storage. 
What is the lifetime of this state and  does it further leak to some system cache?
I suggest to save some flag in the Bundle, which, if present, means the mnemonic is saved and needs to be retrived.
The mnemonic itself can be saved in App static scope using SecretString to erase it as soon as it is not necessary. 
This in not much harder to do, but we have full control of the secret lifetime.

Reply:
The bundle is saved within a system service which can't be accessed by non-system processes _on unrooted devices_
(more info here: https://stackoverflow.com/questions/8091992/where-is-the-bundle-of-onsaveinstancestate-saved)

For more security, we can use a ViewModel instead. It is a bit more code but under has no persistance and is automatically erased when the owning dialog is dismissed (https://developer.android.com/topic/libraries/architecture/saving-states#options)

Reply:
Let's save in the ViewModel, so it is not leaked even to the system.
If we will be able later to advertise it officially, as special care has been taken.